pennylane/math/is_independent.py

```diff
@@ -0,0 +1,358 @@
+        bounds (tuple[int, int]): Limits of the range from which to sample
+
+    Returns:
+        bool: Whether ``func`` returns the same output at randomly
+        chosen points and is numerically independent of its arguments.
+
+    .. warning::
+
+        This function is experimental.
+        As such, it might yield wrong results and might behave
+        slightly differently in distinct ``interface``s for some edge cases.
+        A currently known edge case is the function
+        ``lambda x: x if abs(x) <1e-5 else 0. * x`` at ``x=0.0``.
+
+    The first, analytic test differs per ``interface`` both in its method
+    and its degree of reliability. The respective method is detailed in the
+    docstrings of ``_tf_is_indep_analytic`` for TensorFlow,
+    ``_autograd_is_indep_analytic`` for Autograd and ``_jax_is_indep_analytic``
+    for JAX.
+    The function is then checked numerically to produce constant output at
+    a series of random positions, using ``_is_indep_numerical``.
+
+    .. warning ::
+
+        Currently, no analytical test is available in the PyTorch interface.
+        Only the numerical test ``_is_indep_numerical`` is performed in this case.
+        This is also remarked by a UserWarning.
+
+    .. note ::
+
+        Due to the structure of ``is_independent``, it is possible that it
+        errs on the side of reporting a dependent function to be independent
+        (false positive) but reporting an independent function to be
+        dependent (false negative) is *very* unlikely.
+
+    **Example**
+
+    Consider the (linear) function
+
+    .. code-block:: python
+
+        def lin(x, weights=None):
+            return np.dot(x, weights)
+
+    This function clearly depends on ``x``. We may check for this via
+
+    .. code-block:: pycon
+
+        >>> x = np.array([0.2, 9.1, -3.2])
+        >>> weights = np.array([1.1, -0.7, 1.8])
+        >>> qml.math.is_independent(lin, "autograd", (x,), {"weights": weights})
+        False
+
+    However, the Jacobian will not depend on ``x`` because ``lin`` is a
+    linear function:
+
+    .. code-block:: pycon
+
+        >>> jac = qml.jacobian(lin)
+        >>> qml.math.is_independent(jac, "autograd", (x,), {"weights": weights})
+        True
+
+    A function like ``0.0*x`` will be counted as *dependent* on ``x`` because it does
+    depend on ``x`` *functionally*, even if the value is constant for all ``x``.
+    This means that ``is_independent`` is a stronger test than for constant
+    output functions.
+    """
+    if not interface in {"autograd", "jax", "tf", "torch"}:
+        raise ValueError(f"Unknown interface: {interface}")
+
+    kwargs = kwargs or {}
+
+    if interface == "autograd":
+        if not _autograd_is_indep_analytic(func, *args, **kwargs):
+            return False
+
+    if interface == "jax":
+        if not _jax_is_indep_analytic(func, *args, **kwargs):
+            return False
+
+    if interface == "tf":
```

Comment:
```suggestion
    if interface in ("tf", "tensorflow"):
```
minor, but we sometimes have users do this, and we have started allowing this slowly :)